services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/SqsBatchFunctions.java

```diff
@@ -0,0 +1,333 @@
+            SendMessageBatchRequest batchRequest;
+            if (overrideConfiguration.isPresent()) {
+                batchRequest = SendMessageBatchRequest.builder()
+                                                      .queueUrl(destination)
+                                                      .overrideConfiguration(overrideConfiguration.get())
+                                                      .entries(entries)
+                                                      .build();
+            } else {
+                batchRequest = SendMessageBatchRequest.builder()
+                                                      .queueUrl(destination)
+                                                      .entries(entries)
+                                                      .build();
+            }
+            // TODO: Pass client executor in supplyAsync once an executor is added into the client.
+            return CompletableFuture.supplyAsync(() -> client.sendMessageBatch(batchRequest));
+        };
+    }
+
+    public static BatchResponseMapper<SendMessageBatchResponse, SendMessageResponse> sendMessageResponseMapper() {
+        return sendMessageBatchResponse -> {
+            List<IdentifiableMessage<SendMessageResponse>> mappedResponses = new ArrayList<>();
+            sendMessageBatchResponse.successful()
+                                    .forEach(batchResponseEntry -> {
+                                        String key = batchResponseEntry.id();
+                                        SendMessageResponse response = createSendMessageResponse(batchResponseEntry,
+                                                                                                 sendMessageBatchResponse);
+                                        mappedResponses.add(new IdentifiableMessage<>(key, response));
+                                    });
+            sendMessageBatchResponse.failed()
+                                    .forEach(batchResponseEntry -> {
+                                        String key = batchResponseEntry.id();
+                                        SendMessageResponse response = createSendMessageResponse(batchResponseEntry,
+                                                                                                 sendMessageBatchResponse);
+                                        mappedResponses.add(new IdentifiableMessage<>(key, response));
+                                    });
+            return mappedResponses;
+        };
+    }
+
+    // TODO: The BatchKeyMappers for SQS is not set in stone. Could batch requests some other way.
+    public static BatchKeyMapper<SendMessageRequest> sendMessageBatchKeyMapper() {
+        return request -> {
+            if (request.overrideConfiguration().isPresent()) {
+                return request.queueUrl() + request.overrideConfiguration().get().hashCode();
+            } else {
+                return request.queueUrl();
+            }
+        };
```

Comment:
Same here. `request.overrideConfiguration().map(...).orElse(...)`

Reply:
Refactored code with the suggestion above.